api/src/pcapi/core/search/__init__.py

```diff
@@ -426,61 +426,43 @@ class BookingsCount:
     booking_count_by_ean: dict[str, int]
 
 
-def get_last_x_days_bookings_count(
-    offers: Iterable[offers_models.Offer], days: int = DEFAULT_DAYS_FOR_LAST_BOOKINGS, get_related_eans: bool = False
-) -> BookingsCount:
-    if not FeatureToggle.ALGOLIA_BOOKINGS_NUMBER_COMPUTATION.is_active():
-        return BookingsCount(booking_count_by_offer_id={}, booking_count_by_ean={})
-
-    eans = set()
-    for offer in offers:
-        ean = (offer.extraData or {}).get("ean")
-        if ean is not None:
-            eans.add(ean)
-
+def get_offers_booking_count_by_id(
+    offer_ids: Iterable[int], days: int = DEFAULT_DAYS_FOR_LAST_BOOKINGS
+) -> dict[int, int]:
     offer_booked_since_x_days = (
         offers_models.Offer.query.join(offers_models.Offer.stocks)
         .join(offers_models.Offer.product)
         .join(offers_models.Stock.bookings)
         .filter(
+            offers_models.Offer.id.in_(offer_ids),
+            offers_models.Offer.isActive.is_(True),
             bookings_models.Booking.dateCreated >= datetime.datetime.utcnow() - datetime.timedelta(days=days),
             bookings_models.Booking.status != bookings_models.BookingStatus.CANCELLED,
-            offers_models.Offer.isActive.is_(True),
         )
-    )
-
-    booking_count_by_offer_id = {
-        row[0]: row[1]
-        for row in offer_booked_since_x_days.filter(offers_models.Offer.id.in_(offer.id for offer in offers))
         .group_by(offers_models.Offer.id)
         .with_entities(offers_models.Offer.id, sa.func.count(bookings_models.Booking.id))
-    }
-
-    if get_related_eans:
-        booking_count_by_ean = {
-            row[0]: row[1]
-            for row in offer_booked_since_x_days.filter(offers_models.Offer.extraData["ean"].astext.in_(eans))
-            .group_by(offers_models.Offer.extraData["ean"])
-            .with_entities(offers_models.Offer.extraData["ean"], sa.func.count(bookings_models.Booking.id))
-        }
-
-    return BookingsCount(
-        booking_count_by_offer_id=booking_count_by_offer_id,
-        booking_count_by_ean=booking_count_by_ean if get_related_eans else {},
     )
+    return dict(offer_booked_since_x_days)
 
 
-def get_offer_bookings_count(offer: offers_models.Offer, last_x_days_bookings_count: BookingsCount) -> int:
-    booking_count_by_offer_id = last_x_days_bookings_count.booking_count_by_offer_id
-    booking_count_by_ean = last_x_days_bookings_count.booking_count_by_ean
+def get_all_booking_counts(offers: Iterable[offers_models.Offer]) -> dict[int, int]:
+    offers_with_ean = [
+        offer for offer in offers if offer.product and offer.product.extraData and offer.product.extraData.get("ean")
+    ]
+    offers_without_ean = [
+        offer
+        for offer in offers
+        if not offer.product or not offer.product.extraData or not offer.product.extraData.get("ean")
+    ]
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏
```python
offers_with_ean = []
offers_without_ean = []

for offer in offers: 
  if offer.product and offer.product.extraData and offer.product.extraData.get("ean"):
    offers_with_ean.append(offer)
  else:
    offers_without_ean.append(offer)
```
Pour éviter de dupliquer la condition